lib/policy_machine.rb

```diff
@@ -230,6 +230,32 @@ def batch_find(type:, query: {}, config: {}, &blk)
   end
 
   ##
+  # Search for and iterate over a collection of specified attributes in batches
+  def batch_pluck(type:, query: {}, fields:, config: {}, &blk)
+    return to_enum(__callee__, type: type, query: query, fields: fields, config: config) unless block_given?
+
+    # If the storage adapter implements batch_pluck, delegate
+    if policy_machine_storage_adapter.respond_to?(:batch_pluck)
+      # Fields must include a primary key to avoid ActiveRecord errors
+      fields << :id
+      policy_machine_storage_adapter.batch_pluck(type, query: query, fields: fields, config: config) do |batch|
+        yield batch.to_a
+      end
+    else
+      batch_size = config.fetch(:batch_size, 1)
+
+      method(type.to_s.pluralize).call(query).reduce([]) do |results, pe|
```

Comment:
Or actually, can't you just wrap `batch_find` here, since we have it implemented at this level? Seems like it'd be less code and potentially more efficient.